src/kernels/internalTypes.ts

```diff
@@ -15,6 +15,11 @@ export enum ContributedKernelFinderKind {
 export interface IContributedKernelFinder<T extends KernelConnectionMetadata = KernelConnectionMetadata> {
     status: 'discovering' | 'idle';
     onDidChangeStatus: Event<void>;
+    /**
+     * Last error thrown when listing the kernels.
+     * Use this property to determine if there was an error fetching kernels when there are no kernels listed.
+     */
+    lastError?: Error;

```

Comment:
I don't know the full flow here, but this pattern seems a little suspicious or like it could lead to async issues- like maybe the error should be associated with a particular call to a method or an event.